mllib/src/main/scala/org/apache/spark/mllib/tree/impl/DecisionTreeMetadata.scala

```diff
@@ -107,8 +107,11 @@ private[tree] object DecisionTreeMetadata extends Logging {
       numTrees: Int,
       featureSubsetStrategy: String): DecisionTreeMetadata = {
 
-    val numFeatures = input.take(1)(0).features.size
     val numExamples = input.count()
+    require(numExamples > 0, s"DecisionTree requires size of input RDD > 0, " +
```

Comment:
You should use `isEmpty` rather than count the whole data set. Does this help much? You get an exception either way. Although this makes the message nicer. At the cost of non-trivial extra work.

At this stage wouldn't the size have already had to be positive? have you encountered this in real life?